### PR TITLE
refactor: add RAML Schema Parser

### PR DESCRIPTION
**Pull request by smoya**
**Description**

This PR does the following:

- Ports https://github.com/asyncapi/raml-dt-schema-parser to TypeScript.
- Makes it compliant with the [new interface](https://github.com/asyncapi/parser-js/issues/480) for Schema Parsers (new `validate()` method).

> **Note**
> Schema validation is done through https://github.com/raml-org/webapi-parser. It does not provide a real JSON Path value for validation errors or something that can be really transformed in all cases. By now, path will point to the message payload but we can keep investigating other solutions.

**Related issue(s)**
https://github.com/asyncapi/parser-js/issues/480

### CODE REVIEW

File: src/schema-parser/raml-schema-parser.ts
Comment (smoya) [inline src/schema-parser/raml-schema-parser.ts:25]: Same as https://github.com/asyncapi/parser-js/pull/576#discussion_r943559156

**Review by fmvilas — APPROVED** [review]

File: src/schema-parser/raml-schema-parser.ts
Comment (magicmatatjahu) [reply src/schema-parser/raml-schema-parser.ts:25]: I wrote comment about that in PR for AVRO :) We will handle that in one place.

File: src/schema-parser/raml-schema-parser.ts
Comment (magicmatatjahu) [inline src/schema-parser/raml-schema-parser.ts:55]: In the custom parsers path of single error should point to from perspective of given payload - in other words, payload should be treated as root of error, we will concatenate error path in one place, concatenate path of schema + path of error, so here you should return empty array.

 ```suggestion
      path: [] // Info provided by the RAML parser doesn't provide a real path to the error.
```

**Review by magicmatatjahu — COMMENTED** [review]
Great. I left two comments, please refer to them and we can merge that PR :)

File: src/schema-parser/raml-schema-parser.ts
Comment (smoya) [reply src/schema-parser/raml-schema-parser.ts:55]: Let's say `input.path` value is: `["channels", "myChannel", "publish", "message", "payload"]`, the behaviour will be different that https://github.com/asyncapi/parser-js/pull/578/files#r952869403

Comment (magicmatatjahu) [reply src/schema-parser/raml-schema-parser.ts:55]: But there you operate on error results, in mentioned comment you create mock data for testing. `.path` of input is location on which data you operate some logic, `error.path` is path of the error, but from `.path` point of view - it's a local path that we will merge after validation 😄

Comment (smoya) [reply src/schema-parser/raml-schema-parser.ts:55]: That means, this is also wrong, right? https://github.com/asyncapi/parser-js/blob/next-major/src/schema-parser/asyncapi-schema-parser.ts#L68

Comment (smoya) [reply src/schema-parser/raml-schema-parser.ts]: Changed!

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]
One thing, rest is awesome! :)

File: src/schema-parser/raml-schema-parser.ts
Comment (smoya) [reply src/schema-parser/raml-schema-parser.ts]: Removed!

**Review by magicmatatjahu — APPROVED** [review]
LGTM! 🚀 Should we wait for review of parser's maintainers?

**Comment by smoya** [comment]
We can merge now as we have one approval from a code owner.

**Pull request merged**
